src/query/plan/operator.cpp

```diff
@@ -5986,8 +5986,9 @@ UniqueCursorPtr RollUpApply::MakeCursor(utils::MemoryResource *mem) const {
   return MakeUniqueCursorPtr<RollUpApplyCursor>(mem, *this, mem);
 }
 
-PeriodicCommit::PeriodicCommit(std::shared_ptr<LogicalOperator> &&input, Expression *commit_frequency)
-    : input_(std::move(input)), commit_frequency_(commit_frequency) {}
+PeriodicCommit::PeriodicCommit(std::shared_ptr<LogicalOperator> &&input, Expression *commit_frequency,
+                               DatabaseAccessProtector db_acc)
```

Comment:
Don't put `DatabaseAccessProtector` in the Operator, pass it at execution time using the `ExecutionContext`